plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -1112,11 +1116,86 @@ static int dump_queues(int fd, struct kfd_ioctl_criu_process_info_args *info_arg
 	return ret;
 }
 
+static int dump_events(int fd,
+			struct kfd_ioctl_criu_process_info_args *info_args,
+			CriuKfd *e)
+{
+	int ret = 0, i;
+	struct kfd_ioctl_criu_dumper_args args = {0};
+	struct kfd_criu_event_bucket *ev_buckets;
+	uint8_t *priv_data;
+
+	pr_debug("Dumping %d events\n", info_args->total_events);
+
+	if (!info_args->total_events)
+		return 0;
+
+	args.type = KFD_CRIU_OBJECT_TYPE_EVENT;
+	args.objects_index_start = 0; /* Partial object lists not supported for now */
+
+	args.num_objects = info_args->total_events;
+	args.objects_size = (args.num_objects * sizeof(*ev_buckets)) +
+				info_args->events_priv_data_size;
+
+	args.objects = (uintptr_t)xmalloc(args.objects_size);
+	if (!args.objects) {
+		ret = -ENOMEM;
+		goto exit;
+	}
+
+	ret = kmtIoctl(fd, AMDKFD_IOC_CRIU_DUMPER, &args);
+	if (ret) {
+		pr_perror("amdgpu_plugin: Failed to call dumper (events) ioctl");
+		goto exit;
+	}
+
+	ev_buckets = (struct kfd_criu_event_bucket*)args.objects;
+	/* First private data starts after all buckets */
+	priv_data = (uint8_t *)args.objects + (args.num_objects * sizeof(*ev_buckets));
+
+	e->num_of_events = info_args->total_events;
+
+	ret = allocate_ev_entries(e, args.num_objects);
```

Comment:
Use e->num_of_events instead of args.num_objects.

Reply:
ACK